modules/http-statistics-monitoring/src/main/java/org/wso2/appserver/monitoring/HttpStatValve.java

```diff
@@ -0,0 +1,194 @@
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import javax.servlet.ServletException;
+
+/**
+ * Custom Tomcat valve to Publish server statistics data to Data Analytics Server.
+ */
+public class HttpStatValve extends ValveBase {
+
+    private static final Log LOG = LogFactory.getLog(HttpStatValve.class);
+    private String username = DefaultConfigurationConstants.USERNAME;
+    private String password = DefaultConfigurationConstants.PASSWORD;
+    private String configFileFolder = DefaultConfigurationConstants.CONFIG_FILE_FOLDER;
+    private String url = DefaultConfigurationConstants.URL;
+    private String streamId = DefaultConfigurationConstants.STREAM_ID;
+    private DataPublisher dataPublisher = null;
+    private String appServerHome;
+
+    @Override
+    protected void initInternal() {
+        LOG.debug("The HttpStatValve initialized.");
+        File userDir = new File(System.getProperty("catalina.base"));
+        appServerHome = userDir.getAbsolutePath();
+    }
+
+    @Override
+    public void invoke(Request request, Response response)  {
+
+        if (dataPublisher == null) {
+            try {
+                dataPublisher = getDataPublisher();
+            } catch (ConfigurationException e) {
+                LOG.error("Failed at setting configurations: " + e);
+            }
+        }
+
+        Long startTime = System.currentTimeMillis();
+        try {
+            getNext().invoke(request, response);
+        } catch (IOException | ServletException e) {
+            LOG.error("Invoke failed:" + e);
+        }
+        long responseTime = System.currentTimeMillis() - startTime;
+
+        Event event = null;
+        try {
+            event = EventBuilder.buildEvent(getStreamId(), request, response, startTime, responseTime);
+        } catch (EventBuilderException e) {
+            LOG.error("Creating the Event failed: " + e);
+        }
+        dataPublisher.publish(event);
+
+    }
+
+    //get file path to the file containing Data Agent configuration and properties
+    private String getDataAgentConfigPath() {
+        Path path = Paths.get(appServerHome, getConfigFileFolder(), "data-agent-conf.xml");
+        return path.toString();
+    }
+
+    //instantiate a data publisher to be used to publish data to DAS
+    private DataPublisher getDataPublisher() throws ConfigurationException {
+
+        AgentHolder.setConfigPath(getDataAgentConfigPath());
+        String url = getUrl();
+        DataPublisher dataPublisher;
+
+        try {
+            dataPublisher = new DataPublisher(url, getUsername(), getPassword());
+        } catch (DataEndpointAgentConfigurationException | DataEndpointException | DataEndpointConfigurationException |
+                DataEndpointAuthenticationException | TransportException e) {
+            throw new ConfigurationException("Configuring Data Endpoint Agent failed", e);
```

Comment:
handle exceptions properly with useful messages of each exception. (This was covered during the code review)